gradle/dependencies.gradle

```diff
@@ -62,7 +62,7 @@ def build = [
     errorProneJavac         : "com.google.errorprone:javac:9+181-r4173-1",
     errorProneTestHelpers   : "com.google.errorprone:error_prone_test_helpers:${versions.errorProneApi}",
     checkerDataflow         : "org.checkerframework:dataflow-nullaway:${versions.checkerFramework}",
-    guava                   : "com.google.guava:guava:22.0",
+    guava                   : "com.google.guava:guava:24.1.1-jre",
```

Comment:
A bit hesitant to use `-jre`, but that's what Error Prone does and it's still compatible with our Android monorepo and other projects where we force the Android version.

Reply:
`-jre` is the right thing to use here; I think `-android` should only be used if you need to ship a smaller jar (e.g., with an Android app)

Reply:
The point being that we want NullAway to be able to run when `-android` is provided at runtime, in the absence of `-jre`. But I think the way EP does it is: depend on `-jre`, then make sure to not use APIs that aren't part of both jars? Not 100% sure there.

Reply:
Are you saying in some scenarios developers might be using `-android` as part of their annotation processor path?  I would expect that if `-android` was being used it would only be in the classpath, not the processor path.  I'm unfamiliar with the scenario of using `-android` on the processor path so unsure how much we need to worry about it.

Reply:
Um, now that I think about it, I am not sure if `-jre` is being pulled in for the annotation processor path / plugin path due to EP. It seems to be available in our jar cache, so it's being used for something. I got a bit confused since only `-android` appears as an explicit dependency anywhere in our build files, but I think we are actually getting `-jre` transitively for annotation processors, not forcing `-android` there.